client/core/core.go

```diff
@@ -608,6 +608,19 @@ func (dc *dexConnection) reconcileTrades(srvOrderStatuses []*msgjson.OrderStatus
 		previousStatus := trade.metaData.Status
 		newStatus := order.OrderStatus(srvOrderStatus.Status)
 		trade.metaData.Status = newStatus
+		// If there is an associated cancel order, and we are revising the
+		// status of the targeted order, we can infer the cancel order is done.
+		if trade.cancel != nil {
+			cid := trade.cancel.ID()
+			err := trade.db.UpdateOrderStatus(cid, order.OrderStatusRevoked)
+			if err != nil {
+				dc.log.Errorf("Error updating status in db for cancel order %v to executed", cid)
+			}
+			// Unlink the cancel order from the trade.
+			trade.cancel = nil
+			trade.metaData.LinkedOrder = order.OrderID{}
+		}
```

Comment:
What do you think of refactoring this whole section into a `trackedTrade.deleteCancelOrder()` method, which can then be called from here AND from the existing `trackedTrade.deleteStaleCancelOrder()` method?

Could also call this new `trackedTrade.deleteCancelOrder()` method when a match/nomatch request sets a trade's status to Executed (any associated cancel order would have failed to match and updating the status accordingly should prevent the order from being reloaded from db on restart).

Reply:
Good suggestion to reuse this as `deleteCancelOrder`.  I have it called now from `deleteStaleCancelOrder` and here in the `updateOrder` closure.

Thinking about `nomatch(tradeID)`, I'm hesitant to delete any cancel order from it because if an associated cancel order failed to match there would be a `nomatch(cancelID)`, which does the right thing for the cancel order's status.  So it should be redundant, but might also be undesirable in some way.